### PR TITLE
Apply #120 to 7.x 1.10 release

### PR DESCRIPTION
**Pull request by bondjimbond**
**JIRA Ticket**: 
https://jira.duraspace.org/browse/ISLANDORA-1837
https://jira.duraspace.org/browse/ISLANDORA-2063

# What does this Pull Request do?

Takes the changes from @glewisorg's PR #120 and applies to the 7.x-1.10 branch.

### CODE REVIEW

File: includes/admin.form.inc [hunk 1]
```diff
@@ -16,7 +16,6 @@
  *   The Drupal form definition.
  */
 function islandora_entities_settings($form, &$form_state) {
-
```
Comment (rosiel) [inline includes/admin.form.inc:19]: When doing a Pull Request that's the release branch version of a 7.x Pull request, it's gonna be a pain in the ass if it changes stuff that the other pull request doesn't. This blank line (and the dozen or so others) need to stay, at least for now.

**Review by rosiel — CHANGES_REQUESTED** [review]

File: includes/admin.form.inc (hunk 1 above)
Comment (rosiel) [reply includes/admin.form.inc:19]: I know you've done a lot of work on this, but can we hold on for a moment? I'm not sure if this actually qualifies as a bug fix, and whether we can/should include it in the release. I'm hoping we can discuss this and #120 shortly. Thanks!

**Comment by DiegoPino** [comment]
@rosiel sorry, i'm super confused. How does this not qualify as a bug fix? The error reported on JIRA ticket ISLANDORA-1837 

> Undefined index: mods_name_personal_displayForm_mt in islandora_entities_get_related_pids() (line 240 of /var/www/drupal7/sites/all/modules/islandora_solution_pack_entities/includes/utilities.inc).
> Warning: Invalid argument supplied for foreach() 

it is an actual error, I think we agreed on that during the committers call, so i guess, you mean this one does not fix the cited notice and warning? Can we guide @bondjimbond on how to fix it if not?

**Comment by DiegoPino** [comment]
@bondjimbond 
After some talks with Rosie, i see there are a few issues with this pull:

A) it tries to solve 2 JIRA tickets, one is an improvement (ISLANDORA-1837) and the other a Bug fix (ISLANDORA-2063). We can only merge into the Release if it deals with the Bug fix (ISLANDORA-2023): so all the code that applies the improvement needs to be removed. Sorry, rules are rules. 

B) the bug fix is, according to @rosiel comments, not being addressed. I would ask @rosiel that opened a pull request that [deals with ISLANDORA-1587 which is basically the same issue](https://github.com/Islandora/islandora_solution_pack_entities/pull/123) to help you get the fix done so this pull gets into the release.

In conclusion: 
- remove any reference to 1837 (pull message, title and code)
- get the actual fix into the code. With 1. your own code or 2.  looking at https://github.com/Islandora/islandora_solution_pack_entities/pull/123,  or with @rosiel's help. Whatever makes everyone happier.

2063, if fixed right will also close 1594. 
Once that happens, https://github.com/Islandora/islandora_solution_pack_entities/pull/123 can be closed. I would like to make this one the fix. If we all work together that can be done for sure.

@bryjbrown ping. What do you think? @rosiel @bondjimbond any other concerns?

Thanks a lot, go team 😄

**Comment by DiegoPino** [comment]
@bondjimbond @rosiel @bryjbrown i moved the previous comment from #120  to this. #120 was never intended to be a bug fix, so let's forget about that one please for now and clean that one from comments that attribute that improvement to the bug fix. (correct me of course if that is wrong)

This one claims to fix ISLANDORA-2063 but it seems not (previous comments of Rosie on #120 ) so all my previous comments apply to this one.
If this, with all the comments from @rosiel becomes actually a fix for 2063 (this one, again, not 120!!) then you will have to open an actual pull request against 7.x with the same logic. And this time, please with a full pull request body.

We need to follow some basic rule here:
1) not forging other people's pulls. So if person A) makes a pull request against 7.x, Person B) can not make the same pull request to 7.x-1.10. Attribution is in place in GPL and that contribution is from person A)
2) Not open other pull requests that touch the same code other people are working one. Give the original author a chance to fix it or collaborate. 
3) check for duplicate JIRA tickets. 

Gosh. This whole thing is confusing.




Now i have to go back to 120 and apologies to the author.

**Comment by mjordan** [comment]
@DiegoPino this is not a comment on this PR, but I wonder if we should discuss whether those nice three rules you outline above should be incorporated into https://github.com/Islandora/islandora/wiki/Islandora-Committers-Workflow?

**Comment by DiegoPino** [comment]
@mjordan they are actual non-spoken rules that have kept the open source universe without collapsing into a single atom (a heavy one) for quite a time 😸 I feel they are already implicit in some places (like GPL) or even in our own merge decision. That said, if you feel they are actually OK (because this whole pull mix was a bit confusing and took us quite some time just in organizing, so I could be delusional/dizzy!) I would happy to have them in explicit form and open to discussing and encouraging this.

**Comment by bondjimbond** [comment]
@DiegoPino Sorry, rules I didn't know. :) Shall I just close this PR then?

**Comment by DiegoPino** [comment]
@bondjimbond, sorry, committing sometimes is complicated and keeping track of all the written and sometimes invisible rules is plus complicated.

But, i see some good chances to convert this into a win if you want to:

You could basically make an actual fix for ISLANDORA-2063 converting this pull. Starting by removing the code from https://github.com/Islandora/islandora_solution_pack_entities/pull/120, working with others on the fix and pursue after that the sister 7.x pull based on it. I'm sure you will get help from @rosiel, @bryjbrown, me and any other committer

Or, if you think this already got too complicated, yes. Closing this and helping/reviewing on https://github.com/Islandora/islandora_solution_pack_entities/pull/123 is also an option.

I'm more the "once started, let's finish it" type of coder and I feel this is a great opportunity to actually deal with that type of deal for you. I made so many errors at the beginning, in times where rules were mostly based on observation and i think everyone reading this was there at some time (and we will keep making mistakes for sure!) The end result is quite satisfactory if you allow yourself to make mistakes and get help from all of us 👍 

It is up to you.

**Comment by bondjimbond** [comment]
@DiegoPino After talking with @rosiel it looks like ISLANDORA-2063 is more complicated than just the hard-coded field... I think it's best for me to close this one, let #123 go through, see how that works, and then maybe we decide how to deal with #120?

**Comment by DiegoPino** [comment]
Your call @bondjimbond. feel free to proceed. thanks
